cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -39,26 +43,27 @@ logToProgress mbj l = let
     -- messages until the maximum number of backjumps has been reached. It filters out
     -- and ignores repeated backjumps. If proc reaches the backjump limit, it truncates
     -- the 'Progress' and ends it with the last conflict set. Otherwise, it leaves the
-    -- original success result or replaces the original failure with 'Nothing'.
-    proc :: Maybe Int -> Progress Message a b -> Progress Message (Maybe (ConflictSet QPN)) b
+    -- original result.
+    proc :: Maybe Int -> Log Message b -> Progress Message (Exhaustive, ConflictSet QPN, ConflictMap) b
     proc _        (Done x)                          = Done x
-    proc _        (Fail _)                          = Fail Nothing
+    proc _        (Fail (cs, cm))                   = Fail (Exhaustive, cs, cm)
     proc mbj'     (Step x@(Failure cs Backjump) xs@(Step Leave (Step (Failure cs' Backjump) _)))
       | cs == cs'                                   = Step x (proc mbj'           xs) -- repeated backjumps count as one
-    proc (Just 0) (Step   (Failure cs Backjump)  _) = Fail (Just cs)
+    proc (Just 0) (Step   (Failure cs Backjump)  _) = Fail (NotExhaustive, cs, mempty) -- No final conflict map available
     proc (Just n) (Step x@(Failure _  Backjump) xs) = Step x (proc (Just (n - 1)) xs)
     proc mbj'     (Step x                       xs) = Step x (proc mbj'           xs)
 
-    -- Sets the conflict set from the first backjump as the final error, and records
-    -- whether the search was exhaustive.
-    useFirstError :: Progress Message (Maybe (ConflictSet QPN)) b
-                  -> Progress Message (Bool, Maybe (ConflictSet QPN)) b
+    -- Sets the conflict set from the first backjump as the final error in case of a
+    -- non-exhaustive search.
+    useFirstError :: Progress Message (Exhaustive, ConflictSet QPN, ConflictMap) b
+                  -> Progress Message (Exhaustive, ConflictSet QPN, ConflictMap) b
     useFirstError = replace Nothing
       where
         replace _       (Done x)                          = Done x
-        replace cs'     (Fail cs)                         = -- 'Nothing' means backjump limit not reached.
+        replace _       (Fail (Exhaustive,    cs, cm))    = Fail (Exhaustive, cs, cm)
+        replace cs'     (Fail (NotExhaustive, cs, cm))    = -- Backjump limit not reached.
```

Comment:
Indeed, I'll the `data Exhaustive` to `data Exhaustiveness = Exhaustive | BackjumpLimitReached` so the comment is now unnecessary.